### PR TITLE
Fixes #16: breaking up the project into different files and formatting code better

### LINKED ISSUE #16: [rewrite] Break up into service objects/resources
I don't like approach of just appending functions to the `IntelOwlClient` struct.

This SDK should be created following the approach given in [this](https://medium.com/@marcus.olsson/writing-a-go-client-for-your-restful-api-c193a2f4998c) blog-post, especially the "Service Objects" heading.

This approach is very similar to how we (at Certego) wrote [pydragonfly](https://github.com/certego/pydragonfly) recently using the [django-rest-client](https://github.com/certego/django-rest-client) framework (also written by us).

### PR DESCRIPTION
**Pull request by 0x0elliot**
Please let me know if I can do the tests better and the commenting better. I will implement the feedback. The sooner I know what to improve here, The sooner can I make the right changes and get to implementing other API endpoints.

### CODE REVIEW

**Comment by 0x0elliot** [comment]
I also feel like we should have #14 added in as soon as we can so that we can safely move ahead.

File: gointelowl.go [hunk 1]
```diff
@@ -1,100 +1,75 @@
...
 )
 
 // Struct for IntelOwl.
 // Initiate it using Token, Instance, Certificate
 type IntelOwlClient struct {
+	BaseURL     string
 	Token       string
 	URL         string
 	Certificate string
+	HTTPClient  *http.Client
```
Comment (eshaan7) [inline gointelowl.go:17]: this should not be exposed so `httpClient`.

File: gointelowl.go [hunk 2]
```diff
@@ -1,100 +1,75 @@
...
 	"net/http"
+	"time"
 )
 
 // Struct for IntelOwl.
 // Initiate it using Token, Instance, Certificate
 type IntelOwlClient struct {
+	BaseURL     string
 	Token       string
 	URL         string
```
Comment (eshaan7) [inline gointelowl.go:15]: not needed

File: gointelowl.go [hunk 3]
```diff
@@ -1,100 +1,75 @@
...
-	"log"
 	"net/http"
+	"time"
 )
 
 // Struct for IntelOwl.
 // Initiate it using Token, Instance, Certificate
 type IntelOwlClient struct {
+	BaseURL     string
 	Token       string
```
Comment (eshaan7) [inline gointelowl.go:14]: should not be exposed.

File: gointelowl.go [hunk 4]
```diff
@@ -1,100 +1,75 @@
...
-	httpClient := http.Client{}
-	response, err := httpClient.Do(request)
-	if err != nil {
-		log.Fatalln(err)
-	}
-	log.Println(url, response.StatusCode)
-	return response
+type Response struct {
+	Code    int
+	Data    string
```
Comment (eshaan7) [inline gointelowl.go:22]: should be interface{}

File: gointelowl.go [hunk 5]
```diff
@@ -1,100 +1,75 @@
...
+	Success bool
```
Comment (eshaan7) [inline gointelowl.go:23]: rename to OK

File: jobs.go [hunk 6]
```diff
@@ -0,0 +1,55 @@
+package gointelowl
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+)
+
+type JobTag struct {
```
Comment (eshaan7) [inline jobs.go:10]: not needed

File: jobs.go [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+type Job struct {
+	id                        int     `json:"id"`
+	is_sample                 bool    `json:"is_sample"`
+	observable_name           string  `json:"observable_name"`
+	observable_classification string  `json:"observable_classification"`
+	file_name                 string  `json:"file_name"`
+	file_mimetype             string  `json:"file_mimetype"`
+	status                    string  `json:"status"`
+	tags                      JobTags `json:"tags"`
```
Comment (eshaan7) [inline jobs.go:26]: replace w/ Tags[]

File: jobs.go [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+	process_time              int     `json:"process_time"`
+	no_of_analyzers_executed  string  `json:"no_of_analyzers_executed"`
+	no_of_connectors_executed string  `json:"no_of_connectors_executed"`
+}
+
+type JobList []Job
```
Comment (eshaan7) [inline jobs.go:32]: redundant, just use []Job

**Pull request closed without merging**